### DISCUSSION TITLE
spectral whitening

### DISCUSSION THREAD

**Discussion by John8974** [ROBelgium/MSNoise#256] [Q&A] [answered]
I try to do spectral whitening through msnoise1.6 and have some questions.

I saw a reference that bandpass filter as 0.04-1.25Hz and spectral whitening as 0.05-1Hz.

In MSNoise, there are parameters related to whitening, which are 'whitening' and 'whitening_type', but I am not sure the way to assign the whitening frequency (0.05-1Hz).

The definition of whitening is related to autocorrelation, none, or components not the frequency.

I saw the table definition about low and high filter, 'the lower/upper frequency bound of the whiten function', but I guess it is not the parameter related to whitening but just a filter.

So could you please let me know the way to assign whitening frequency?

**Comment by ThomasLecocq** [comment, marked as answer]
Hi,
The whitening frequencies are indeed defined in the filter's parameters as "low" and "high", here 0.05 and 1.0 Hz, respectively. They act both as a bandpass within those bounds, and a whitening (all to 1.0, or PSD, or else depending on whitening_type).

Thomas

**Reply by John8974** [reply]
Oh I see. Then if I want to assign bandpass filter and whitening frequency differently, how can I control it?

**Reply by ThomasLecocq** [reply]
it's not possible with msnoise currently, and actually I don't see why you'd want to bandpass and whiten in different frequency bands, what happens to the non-whitened part of the spectrum then ?

In the future, it could (still thinking about how to implement it) to correlate in a wideband and later compute dv/v in shorter bands

**Reply by John8974** [reply]
I am trying to follow some references related to effects of earthquakes on ambient noise and in some reference, they used different bandpass filter, spectral whitening frequency, and mwcs_low/high frequency.

I am not sure the reason they set all frequencies differently but know how to handle bandpass filter and mwcs_low/high frequency, so that is why I wondered the way to control spectral whitening.

Now I understand the 'low' and 'high' filter act as both of bandpass and whitening. Thank you for your help.

p.s. I saw the psd plot. If I want to plot it, do I need to set the whitening type as psd?

**Reply by ThomasLecocq** [reply]
What do you mean "the psd plot" ? The psd-spectrogram ? That is generated using the latest msnoise dev version from github (using `msnoise qc compute_psd` etc), and has nothing to do with PSD-Whitening (which is just another method for whitening the spectra)

**Reply by John8974** [reply]
Yes the psd-spectrogram. I got it.
I am using MSNoise 1.6.1 and when I do `msnoise qc compute_psd`, the error (No such command 'qc') occurs. And when using `'msnoise',` there is no command for it. Is it possible to use the command in MSNoise 1.6.1? Or for that, is there anything that I should install or do something ? If it is possible, it would be useful for analyzing data.

**Reply by ThomasLecocq** [reply]
no, as I said above, the PSD stuff is only available in the development version, i.e. here on github. You'll want to install msnoise from source (=here), but be aware that this dev version is **not** compatible with the 1.6.2 version (changes in the database schema, new config elements, etc).

**Reply by John8974** [reply]
I see. Thank you so much for your help!!
